### PR TITLE
KT-867: Update ODA modules diagram

### PR DESCRIPTION
**Pull request by teemkata**
Add oda-esb, oda-notification-service, oda-cds-service, oda-analytics-service
and dashboard components to diagram and add the descriptions of the components
to README.md.

### CODE REVIEW

File: modules.plantuml [hunk 1]
```diff
@@ -9,30 +9,46 @@ endheader
...
 }
 
 folder "External systems" as ext {
     component "Suomi.fi" as suomifi
     component AVPH
 }
 
-ob --> suomifi
+idp --> suomifi
+esb <--> suomifi
```
Comment (jokimaki) [inline modules.plantuml]: In this diagram suomi.fi has referred to identification service. You should split this to suomi.fi identification (http://tunnistus.suomi.fi) where idp connects to and suomi.fi Data Exchange Layer (X-Road) where esb connects to.

File: modules.plantuml [hunk 2]
```diff
@@ -9,30 +9,46 @@ endheader
...
+    dashboard --> oas
 }
 
 folder "External systems" as ext {
     component "Suomi.fi" as suomifi
     component AVPH
 }
 
-ob --> suomifi
+idp --> suomifi
```
Comment (jokimaki) [inline modules.plantuml]: idp connects to phr too

Comment (teemkata) [reply modules.plantuml]: Fixed.

File: modules.plantuml (hunk 1 above)
Comment (teemkata) [reply modules.plantuml]: Fixed.

**Pull request merged**